src/main/java/teammates/ui/webapi/action/DeleteInstructorAction.java

```diff
@@ -31,9 +40,60 @@ public ActionResult execute() {
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
 
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, instructorId);
+        if (instructor == null) {
+            CourseAttributes courseAttributes = logic.getCourse(courseId);
+            if (courseAttributes == null) {
+                return new JsonResult("No course with id: " + courseId, HttpStatus.SC_NOT_FOUND);
+            }
+            return new JsonResult("No instructor with id: " + instructorId, HttpStatus.SC_NOT_FOUND);
+        }
+
+        // Deleting last instructor from the course is not allowed if you're not the admin
+        if (userInfo.isInstructor && !hasAlternativeInstructor(courseId, instructor.email)) {
+            return new JsonResult("The instructor you are trying to delete is the last instructor in the course. "
+                    + "Deleting the last instructor from the course is not allowed.", HttpStatus.SC_BAD_REQUEST);
+        }
+
         logic.deleteInstructor(courseId, instructor.email);
 
-        return new JsonResult("Instructor is successfully deleted.", HttpStatus.SC_OK);
+        return new JsonResult("OK", HttpStatus.SC_OK);
     }
 
+    /**
+     * Returns true if there is at least one joined instructor (other than the instructor to delete)
+     * with the privilege of modifying instructors and at least one instructor visible to the students.
+     *
+     * @param courseId                Id of the course
+     * @param instructorToDeleteEmail Email of the instructor who is being deleted
+     */
+    private boolean hasAlternativeInstructor(String courseId, String instructorToDeleteEmail) {
+
+        List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
+        boolean isAlternativeInstructorWithModifyInstructorPermissionPresent = false;
```

Comment:
I feel this variable name is far too long! You should be able to shorten it (its almost the whole line :O)